### PR TITLE
Feature/log submission keys

### PR DESCRIPTION
**Pull request by mibrasap**
## Description

Bring the latest changes regarding checking and logging of the submitted diagnosis keys to feature/federation

### CODE REVIEW

File: common/persistence/src/main/java/app/coronawarn/server/common/persistence/domain/DiagnosisKey.java [hunk 1]
```diff
@@ -236,4 +236,20 @@ public int hashCode() {
...
+  @Override
+  public String toString() {
+    return "DiagnosisKey{"
+        + "keyData=HIDDEN"
+        + ", rollingStartIntervalNumber=" + rollingStartIntervalNumber
+        + ", rollingPeriod=" + rollingPeriod
+        + ", transmissionRiskLevel=" + transmissionRiskLevel
+        + ", submissionTimestamp=" + submissionTimestamp
+        + ", consentToFederation=" + consentToFederation
+        + ", originCountry='" + originCountry + '\''
```
Comment (hilmarf) [inline common/persistence/src/main/java/app/coronawarn/server/common/persistence/domain/DiagnosisKey.java:249]: if you need to add ' and use string+ anyhow, why don't use "'" then? reading escape sequences, is often a hard task

**Review by hilmarf — DISMISSED** [review]

File: common/persistence/src/main/java/app/coronawarn/server/common/persistence/domain/DiagnosisKey.java (hunk 1 above)
Comment (mibrasap) [reply common/persistence/src/main/java/app/coronawarn/server/common/persistence/domain/DiagnosisKey.java]: Thanks for the hint, it's fixed

File: services/submission/src/main/java/app/coronawarn/server/services/submission/controller/SubmissionController.java [hunk 2]
```diff
@@ -145,6 +159,27 @@
     return diagnosisKeys;
   }
 
+  private void checkDiagnosisKeysStructure(List<DiagnosisKey> diagnosisKeys) {
+    diagnosisKeys.sort(Comparator.comparing(DiagnosisKey::getRollingStartIntervalNumber));
+    String keysString = Arrays.toString(diagnosisKeys.toArray());
+    Predicate<DiagnosisKey> hasRiskLevel6 = diagnosisKey -> diagnosisKey.getTransmissionRiskLevel() == 6;
+
+    if (diagnosisKeys.stream().noneMatch(hasRiskLevel6)) {
```
Comment (EugenM-SAP) [inline services/submission/src/main/java/app/coronawarn/server/services/submission/controller/SubmissionController.java:167]: the method name says checkDiagnosisKeysStructure which is more general. But in the implementation only risk 6 keys are being checked. Shouldn't we have logging for all types risk levels which would be expected in the payloads but are missing?

**Review by hilmarf — APPROVED** [review]

**Review by hilmarf — APPROVED** [review]

File: services/submission/src/main/java/app/coronawarn/server/services/submission/controller/SubmissionController.java (hunk 2 above)
Comment (mibrasap) [reply services/submission/src/main/java/app/coronawarn/server/services/submission/controller/SubmissionController.java:169]: The number of keys to be submitted depends on the time how long the app is installed on the phone. Hence, the number of keys differs, but the key with transmission risk level 6 is the one from the day before submission and should always be present.

Comment (EugenM-SAP) [reply services/submission/src/main/java/app/coronawarn/server/services/submission/controller/SubmissionController.java:169]: I understand. It's worth adding a javadoc with your explanation

**Review by EugenM-SAP — APPROVED** [review]

**Review by EugenM-SAP — DISMISSED** [review]

File: services/submission/src/main/java/app/coronawarn/server/services/submission/controller/SubmissionController.java (hunk 2 above)
Comment (mibrasap) [reply services/submission/src/main/java/app/coronawarn/server/services/submission/controller/SubmissionController.java:179]: done ;)

**Review by hilmarf — APPROVED** [review]

**Review by EugenM-SAP — APPROVED** [review]

**Pull request merged**